src/CLIPreChecker/CLIPrechecker.ts

```diff
@@ -0,0 +1,58 @@
+import chalk from 'chalk'
+import { spawn } from 'child_process'
+import { join } from 'path'
+import semver from 'semver'
+import { CLIPrecheckerStore, ICLIPrecheckerStore } from './CLIPrecheckerStore'
```

Comment:
Yes, it's common on OOP languages like dotnet, it's even used on VScode codebase, which is in typescript:

https://github.com/microsoft/vscode/blob/dc1e7871b8dd987ca2b25b9d983c2d31ec9039d7/src/vs/platform/extensions/common/extensions.ts#L14-L34

Reply:
I though of trying to decouple more the implementation of something from its interface. This way we could easily implement a `CLIPrecheckerStoreMock` that implements `ICLIPrecheckerStore` for testing purposes, and since `CLIPrechecker` expects an `ICLIPrecheckerStore` it would accept the mocked one.